editor/src/uuiui/radix-components.tsx

```diff
@@ -0,0 +1,99 @@
+import { styled } from '@stitches/react'
+import React from 'react'
+import { colorTheme } from './styles/theme'
+import { Icons } from './icons'
+import { when } from '../utils/react-conditionals'
+
+const RadixItemContainer = styled(RadixDropdownMenu.Item, {
+  minWidth: 128,
+  padding: '4px 8px',
+  display: 'flex',
+  gap: 12,
+  justifyContent: 'space-between',
+  alignItems: 'center',
+  cursor: 'pointer',
+  color: colorTheme.contextMenuForeground.value,
+  '&[data-highlighted]': {
+    backgroundColor: colorTheme.contextMenuHighlightBackground.value,
+    color: colorTheme.contextMenuForeground.value,
+    borderRadius: 6,
+  },
+})
+
+const RadixDropdownContent = styled(RadixDropdownMenu.Content, {
+  padding: '6px 8px',
+  flexDirection: 'column',
+  backgroundColor: colorTheme.contextMenuBackground.value,
+  borderRadius: 6,
+  display: 'grid',
+  gridTemplateRows: '1fr auto',
+})
+
+export interface DropdownMenuItem {
+  id: string
+  label: React.ReactNode
+  shortcut?: string
+  icon?: React.ReactNode
+  checked?: boolean
+  onSelect: () => void
+}
+
+export interface DropdownMenuProps {
+  opener: (open: boolean) => React.ReactNode
+  items: DropdownMenuItem[]
+  sideOffset?: number
+  alignOffset?: number
+}
+
+export const DropdownMenu = React.memo<DropdownMenuProps>((props) => {
+  const stopPropagation = React.useCallback((e: React.KeyboardEvent) => {
+    const hasModifiers = e.altKey || e.metaKey || e.shiftKey || e.ctrlKey
+    if (!hasModifiers) {
+      e.stopPropagation()
+    }
+  }, [])
+  const onEscapeKeyDown = React.useCallback((e: KeyboardEvent) => e.stopPropagation(), [])
+
+  const [open, onOpen] = React.useState(false)
+
+  const shouldShowCheckboxes = props.items.some((i) => i.checked != null)
+  const shouldShowIcons = props.items.some((i) => i.icon != null)
+
+  return (
+    <RadixDropdownMenu.Root open={open} onOpenChange={onOpen}>
+      <RadixDropdownMenu.Trigger style={{ background: 'none', border: 'none' }}>
+        {props.opener(open)}
+      </RadixDropdownMenu.Trigger>
+      <RadixDropdownMenu.Portal>
+        <RadixDropdownContent
+          onKeyDown={stopPropagation}
+          onEscapeKeyDown={onEscapeKeyDown}
+          sideOffset={props.sideOffset}
+          collisionPadding={{ top: -4 }}
+          align='start'
+          alignOffset={props.alignOffset}
+        >
+          {props.items.map((item) => (
+            <RadixItemContainer key={item.id} onSelect={item.onSelect}>
+              <div style={{ display: 'flex', gap: 4, alignItems: 'center' }}>
+                {when(
+                  shouldShowCheckboxes,
+                  <div style={{ opacity: item.checked ? 1 : 0 }}>
+                    <Icons.Checkmark color='white' />
+                  </div>,
+                )}
+                {when(
+                  shouldShowIcons,
+                  <div style={{ opacity: item.icon == null ? 0 : 1 }}>{item.icon}</div>,
+                )}
```

Comment:
minor - but we can maybe merge these, since a checkmark is just a private case of an icon

Reply:
<img width="204" alt="image" src="https://github.com/concrete-utopia/utopia/assets/2226774/fb03116b-783b-410e-9893-9d0eff68c11e">

this needs to make space for a checkmark and an icon separately

Reply:
yes, got it :)